src/Random/Random.dfy

```diff
@@ -0,0 +1,22 @@
+/*******************************************************************************
+ *  Copyright by the contributors to the Dafny Project
+ *  SPDX-License-Identifier: MIT
+ *******************************************************************************/
+
+module {:options "-functionSyntax:4"} Dafny.Random {
+
+    // Return an arbitrary boolean value.
+    predicate {:extern "DafnyLibraries.Random", "nextBool"} nextBool()
```

Comment:
Soundness issue: none of these can be predicates/functions since they are nondeterminisitic. Otherwise Dafny can prove false:

```dafny
if nextReal() != nextReal() {
  print 1/0;    // Verifies but crashes at runtime almost 100% of the time
}
```


```

Reply:
Credit to @seebees for spotting this while I was distracted by other things. :P

Reply:
Team effort, came in this AM to make sure this was on the PR, thanks!

Reply:
One can make them functions as long as they take a `seed` input (ideally an opaque type that can only be created with a method), which is how all of this is handled in e.g. Haskell.